### PR TITLE
Validate aggregation method is either sum or count

### LINKED ISSUE #1137: Bad aggregation method for overview dataview
From logs:
```
[2019-11-07 13:50:44.067] [FATAL] [default] - TypeError: aggregationFnQueryTpl[this.aggregation] is not a function
    at Aggregation.getAggregationSql (/home/ubuntu/www/node-windshaft/releases/20191029111604/lib/models/dataview/overviews/aggregation.js:223:51)
    at Aggregation.sql (/home/ubuntu/www/node-windshaft/releases/20191029111604/lib/models/dataview/overviews/aggregation.js:191:40)
    at /home/ubuntu/www/node-windshaft/releases/20191029111604/lib/models/dataview/overviews/aggregation.js:127:18
    at psql.query (/home/ubuntu/www/node-windshaft/releases/20191029111604/lib/models/dataview/base.js:75:13)
    at queryArray (/home/ubuntu/www/node-windshaft/releases/20191029111604/node_modules/cartodb-psql/lib/psql.js:359:24)
    at Result.client.query (/home/ubuntu/www/node-windshaft/releases/20191029111604/node_modules/cartodb-psql/lib/psql.js:321:24)
    at Result.Query.handleReadyForQuery (/home/ubuntu/www/node-windshaft/releases/20191029111604/node_modules/pg/lib/query.js:144:10)
    at Connection.<anonymous> (/home/ubuntu/www/node-windshaft/releases/20191029111604/node_modules/pg/lib/client.js:184:19)
    at Connection.emit (events.js:189:13)
    at Socket.<anonymous> (/home/ubuntu/www/node-windshaft/releases/20191029111604/node_modules/pg/lib/connection.js:135:12)
```

We should validate the aggregation method.

### PR DESCRIPTION
**Pull request by esloho**
Fixes #1137 

The method `checkOptions` in `models/dataview/aggregation` already validates that the necessary options are included and that the aggregation methods exist. However, the overview use case has a more restricted list of valid methods (it only accepts sum or count) and thus the existing validation wasn't enough in here.

Extra validation has been added for the overview together with some tests to validate that all cases are covered by one of the validation methods.

### CODE REVIEW

File: test/acceptance/dataviews/overviews-test.js [hunk 1]
```diff
@@ -665,6 +665,194 @@ describe('dataviews using tables with overviews', function() {
...
+                                query: 'select * from test_table_overviews'
+                            }
+                        },
+                        {
+                            id: 'data-source-special-float-values',
+                            type: 'source',
+                            params: {
+                                query: 'select * from test_special_float_values_table_overviews'
+                            }
+                        }
```
[the remark this answers is not in the document]
Comment (esloho) [reply test/acceptance/dataviews/overviews-test.js]: Tests fail without a full config map, so I put it back :S

File: test/acceptance/dataviews/overviews-test.js [hunk 2]
```diff
@@ -665,6 +665,194 @@ describe('dataviews using tables with overviews', function() {
...
+                            type: 'mapnik',
+                            options: {
+                                sql: 'select * from test_special_float_values_table_overviews',
+                                cartocss: '#layer { marker-fill: red; marker-width: 32; marker-allow-overlap: true; }',
+                                cartocss_version: '2.3.0',
+                                source: {
+                                    id: 'data-source-special-float-values'
+                                }
+                            }
+                        }
```
[the remark this answers is not in the document]
Comment (esloho) [reply test/acceptance/dataviews/overviews-test.js]: Same than before, I tried to take each part individually (layers and analyses) but tests fail if removing any of them.

[the remark this answers is not in the document]
Comment (dgaubert) [reply test/acceptance/dataviews/overviews-test.js]: what error are you getting?

[the remark this answers is not in the document]
Comment (esloho) [reply test/acceptance/dataviews/overviews-test.js]: Taking `layers` as example:

When removing the object `layers` entirely I get a syntax error "Unexpected token in JSON". If I keep the `layers` object as an entity list then it says "Unexpected keys found in redis"

[the remark this answers is not in the document]
Comment (esloho) [reply test/acceptance/dataviews/overviews-test.js]: Sorry, I missunderstood the lines that should be removed :D

**Pull request merged**